File: Controls/Converters/DecimalConverter.vb
Imports System.Globalization
Imports System.Collections.ObjectModel

Public NotInheritable Class DecimalConverter
  Inherits Control
  Implements IValueConverter

  Public Property DecimalPositions As Integer = 0
  Public Property IncludeComma As Boolean = True
  Public Property OptionalHeader As String = String.Empty


  Public Function Convert(value As Object, targetType As Type, parameter As Object, culture As CultureInfo) As Object Implements IValueConverter.Convert
    If Not IsNumeric(value) Then Return String.Empty

    Dim FormatString As String = If(IncludeComma, "#,##0", "0")
    If DecimalPositions > 0 Then FormatString &= "." & StrDup(DecimalPositions, "0"c)

    Return $"{OptionalHeader} {CType(value, Decimal).ToString(FormatString)}"

  End Function

  Public Function ConvertBack(value As Object, targetType As Type, parameter As Object, culture As CultureInfo) As Object Implements IValueConverter.ConvertBack
    Throw New NotImplementedException()
  End Function
End Class

File: CSharpControls/Converters/Instances/DecimalConverter.cs
using Microsoft.VisualBasic;
using System;                  
using System.Globalization; 
using System.Windows.Controls;
using System.Windows.Data;

namespace CSharpControls.Converters.Instances
{
  public sealed class DecimalConverter : Control, IValueConverter
  {
    public int DecimalPositions { get; set; } = 0;
    public bool IncludeComma { get; set; } = true;
    public string OptionalHeader { get; set; } = string.Empty;


    public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
    {
      if (!(Information.IsNumeric(value)))
        return string.Empty;

      string FormatString = IncludeComma ? "#,##0" : "0";
      if (DecimalPositions > 0)
        FormatString += "." + Strings.StrDup(DecimalPositions, '0');
                                               
      return $"{OptionalHeader} {System.Convert.ToDecimal(value).ToString(FormatString)}";
    }

    public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
    {
      throw new NotImplementedException();
    }
  }
}
